source/directives/uiTreeNode.js

```diff
@@ -63,7 +63,7 @@
             });
 
             scope.$watch(attrs.expandOnHover, function(val) {
-              if ((typeof val) == 'boolean') {
+              if ((typeof val) == 'boolean' || (typeof val) == 'number') {
```

Comment:
This wasn't the regression (that was in #806), but this PR was involved in the release.